internal/controller/projectdevelopmentstream_controller.go

```diff
@@ -82,6 +88,11 @@ func (r *ProjectDevelopmentStreamReconciler) Reconcile(ctx context.Context, req
 	log = log.WithValues("PDS name", pds.ObjectMeta.Name)
 	log = muxr.NewMuxLogger(log, eventr.NewEventr(r.Recorder, &pds))
 
+	// Set initial condition
+	if err := r.setReadyCondition(ctx, &pds, metav1.ConditionUnknown, "Reconciling", "Reconciling ProjectDevelopmentStream"); err != nil {
+		log.Error(err, "Failed to update status")
```

Comment:
instead of adding multipel log calls here to log each time we fail to set the status condition, we can do the logging from the setReadyCondition function and keep the code here cleaner. We can pass the logger around by putting it into the context, there seem to be a convention of doing that.